lambdas/initialise-ipv-session/src/test/java/uk/gov/di/ipv/core/initialiseipvsession/InitialiseIpvSessionHandlerTest.java

```diff
@@ -80,33 +116,50 @@ class InitialiseIpvSessionHandlerTest {
     @Mock private JarValidator mockJarValidator;
     @Mock private AuditService mockAuditService;
     @Mock private UserIdentityService mockUserIdentityService;
+    @Mock private VerifiableCredentialJwtValidator mockVerifiableCredentialJwtValidator;
+    @Mock private VerifiableCredentialService mockVerifiableCredentialService;
     @InjectMocks private InitialiseIpvSessionHandler initialiseIpvSessionHandler;
 
+    @Captor private ArgumentCaptor<SignedJWT> signedJWTArgumentCaptor;
+
     private final ObjectMapper objectMapper = new ObjectMapper();
 
+    private static Map<String, Object> jarClaims;
+    private static SignedJWT inheritedIdentitySignedJwt;
     private static SignedJWT signedJWT;
     private static JWEObject signedEncryptedJwt;
-    private static IpvSessionItem ipvSessionItem;
+    private static @Spy IpvSessionItem ipvSessionItem;
     private static ClientOAuthSessionItem clientOAuthSessionItem;
 
     @BeforeAll
     static void setUp() throws Exception {
-        JWTClaimsSet claimsSet =
-                new JWTClaimsSet.Builder()
-                        .expirationTime(new Date(Instant.now().plusSeconds(1000).getEpochSecond()))
-                        .issueTime(new Date())
-                        .notBeforeTime(new Date())
-                        .subject("test-user-id")
-                        .audience("test-audience")
-                        .issuer("test-issuer")
-                        .claim("response_type", "code")
-                        .claim("redirect_uri", "http://example.com")
-                        .claim("state", "test-state")
-                        .claim("client_id", "test-client")
-                        .claim("vtr", List.of("Cl.Cm.P2", "Cl.Cm.PCL200"))
-                        .build();
-
-        signedJWT = new SignedJWT(new JWSHeader(JWSAlgorithm.ES256), claimsSet);
+        inheritedIdentitySignedJwt = getInheritedIdentityJWT();
+        jarClaims = new HashMap<>();
+        jarClaims.put(EXPIRATION_TIME, new Date(Instant.now().plusSeconds(1000).getEpochSecond()));
+        jarClaims.put(ISSUED_AT, new Date());
+        jarClaims.put(NOT_BEFORE, new Date());
+        jarClaims.put(SUBJECT, "test-user-id");
+        jarClaims.put(AUDIENCE, "test-audience");
+        jarClaims.put(ISSUER, "test-issuer");
+        jarClaims.put(RESPONSE_TYPE, "code");
+        jarClaims.put(REDIRECT_URI, "https://example.com");
+        jarClaims.put(STATE, "test-state");
+        jarClaims.put(CLIENT_ID, "test-client");
+        jarClaims.put(VTR, List.of("Cl.Cm.P2", "Cl.Cm.PCL200"));
+        jarClaims.put(
+                CLAIMS,
+                new JarClaims(
```

Comment:
Is this allowed - can we put a custom class here? I assume it works, but strictly speaking I think these should be maps.

Reply:
Is there a reason we wouldn't want to do it like this:
```
         JWTClaimsSet claimsSet =
                new JWTClaimsSet.Builder()
                        .expirationTime(new Date(Instant.now().plusSeconds(1000).getEpochSecond()))
                        .issueTime(new Date())
                        .notBeforeTime(new Date())
                        .subject("test-user-id")
                        .audience("test-audience")
                        .issuer("test-issuer")
                        .claim("response_type", "code")
                        .claim("redirect_uri", "http://example.com")
                        .claim("state", "test-state")
                        .claim("client_id", "test-client")
                        .claim("vtr", List.of("Cl.Cm.P2", "Cl.Cm.PCL200"))
                        .build();
```

Reply:
I think it's because there are some tests that might modify these more easily. It might be neater to have a 'default' builder which can be appended to somehow. I haven't thought through all the details though...